runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/kotlin/kotlinClasspathUtils.kt

```diff
@@ -5,25 +5,12 @@ import org.gradle.api.file.FileCollection
 import org.jetbrains.dokka.gradle.isAndroidTarget
 import org.jetbrains.kotlin.gradle.plugin.KotlinSourceSet
 import org.jetbrains.kotlin.gradle.plugin.mpp.AbstractKotlinNativeCompilation
-import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinCommonCompilation
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
-val Project.isHMPPEnabled
-    // [KotlinCommonCompilation.isKlibCompilation] is internal, so we use this
-    get() = (this.findProperty("kotlin.mpp.enableGranularSourceSetsMetadata") as? String)?.toBoolean() ?: false
-
 internal fun Project.classpathOf(sourceSet: KotlinSourceSet): FileCollection {
     val compilations = compilationsOf(sourceSet)
     return if (compilations.isNotEmpty()) {
         compilations
-            /**
-             * If the project has enabled Compatibility Metadata Variant (produces legacy variant),
-             * we don't touch it due to some dependant library
-             * might be published without Compatibility Metadata Variant.
-             * Dokka needs only HMPP variant
-             * Ignore [org.jetbrains.kotlin.gradle.plugin.mpp.KotlinCommonCompilation]  for `commonMain`  sourceSet with name `main`
-             */
-            .filterNot { compilation -> isHMPPEnabled && compilation is KotlinCommonCompilation && compilation.name == "main" }
             .map { compilation -> compilation.compileClasspathOf(project = this) }
             .reduce { acc, fileCollection -> acc + fileCollection }
```

Comment:
There's a check for `if (compilations.isNotEmpty())` above, line 12. If the

This is actually why I moved the filter out of here -- to avoid a second `isEmpty()` check